src/main/java/seedu/address/model/person/UniquePersonList.java

```diff
@@ -98,6 +102,29 @@ public void setPersons(List<Person> persons) {
         internalList.setAll(persons);
     }
 
+    /**
+     * Updates the tag in every contact.
+     * @param oldTag The old tag to be changed.
+     * @param newTag The new tag that is changed into.
+     */
+    public void changeRelevantPersonTag(Tag oldTag, Tag newTag) {
+        for (int i = 0; i < internalList.size(); i++) {
+            Person oldPerson = internalList.get(i);
```

Comment:
Perhaps there should be guard statements here, similar to what you have done for the other methods.